controller/src/main/java/io/pravega/controller/server/rpc/auth/AuthHandlerManager.java

```diff
@@ -136,7 +136,7 @@ public boolean authorize(String resource, Principal principal, String credential
      * to inject and register custom auth handlers. Also, this method is idempotent.
      *
      * @param authHandler the {@code AuthHandler} implementation to register
-     * @Throws NullPointerException {@code authHandler} is null
+     * @throws NullPointerException {@code authHandler} is null
```

Comment:
Personally, I am of the opinion (in agreement with your comment) that an `IllegalArgumentException` - and not `NullPointerException` - should be thrown when an argument is illegal (including the case where `null` is not allowed). `NullPointerException` on the other typically indicates that a variable inappropriately turned out to be a `null`, such as when a `null` is dereferenced. 

There are many who strongly advocate using a NullPointerException instead, for asserting that 'null' isn't allowed as an argument. Joshua Bloch in his famous book "Effective Java" states the following: 
> Arguably, every erroneous method invocation boils down to an illegal argument or state, but other exceptions are standardly used for certain kinds of illegal arguments and states. If a caller passes null in some parameter for which null values are prohibited, convention dictates that NullPointerException be thrown rather than IllegalArgumentException. Similarly, if a caller passes an out-of-range value in a parameter representing an index into a sequence, IndexOutOfBoundsException should be thrown rather than IllegalArgumentException.

A lot of code in our codebase throws NullPointerException to indicate that `null` is not allowed. For example, io.pravega.common.Exceptions.checkNotNullOrEmpty(), which is widely used, throws a NullPointerException if the argument is null. Similarly, `com.google.common.base.checkNotNull() throws a NullPointerException to reject a null input.